source/slang/slang-emit-cpp.cpp

```diff
@@ -2526,6 +2547,45 @@ void CPPSourceEmitter::emitPreModuleImpl()
     }
 }
 
+/* virtual */void CPPSourceEmitter::emitFuncDecorationsImpl(IRFunc* func)
+{
+    // Specially handle export, as we don't want to emit it multiple times
```

Comment:
We should probably have logic elsewhere to ensure that we don't redundantly add decorations.